packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -24,7 +55,54 @@ export class Snapshot {
     // TODO: CDX-347 Display Expanded preview
   }
 
-  async delete() {
-    // TODO: CDX-359: Delete snapshot once previewed
+  private get latestReport(): ResourceSnapshotsReportModel {
+    if (this.model.reports === undefined) {
+      throw new Error(
+        `No detailed report found for the snapshot ${this.snapshotId}`
+      );
+    }
+    return this.model.reports.slice(-1)[0];
+  }
+
+  private isValid() {
+    const {resultCode, status} = this.latestReport;
+    return (
+      status === ResourceSnapshotsReportStatus.Completed &&
+      resultCode === ResourceSnapshotsReportResultCode.Success
+    );
+  }
+
+  private async refreshSnapshotData() {
+    this.model = await this.snapshotClient.get(this.model.id, {
+      includeReports: true,
+    });
+  }
+
+  private async isIdle() {
+    await this.refreshSnapshotData();
+    return new Promise<void>((resolve, reject) => {
+      if (
+        [
+          ResourceSnapshotsReportStatus.Aborted,
+          ResourceSnapshotsReportStatus.Completed,
+        ].includes(this.latestReport.status)
+      ) {
+        resolve();
+      } else {
+        reject();
+      }
+    });
+  }
```

Comment:
Mhhh. Not sure here, what `isIdle` for you in layman terms?

Reply:
Meaning that snapshot is not in a pending state.
In the platform, that pending state is translated with a loading animation (see image).
![image](https://user-images.githubusercontent.com/12199712/120547717-7ee46800-c3bf-11eb-9487-78bd66abf2c7.png)

The role of that `isIdle` function is to keep fetching the snapshot status until the report indicates a failure or a success.
But I agree, we can change that name :P

Reply:
> Meaning that the snapshot is not being processed by some kind of operation (validation, deployment, being applied). That it had reached a finish state.

Is there a defined status for that in `ResourceSnapshotsReportStatus` ?

The thing that I found odd tbh is that `isIdle` resolves if the status is Aborted or Completed.

I interpret that as `isIdle` is 'true' if the status is Completed or Aborted. Which is strange imho

also resolve, reject, meh. I think a resolve w/ a boolean would be more semantically correct with the `isXX` name.

Reply:
`isIdle` doesn't mean the snapshot is valid. It only means that it has settled from its last operation. We ca also go the other way around and have a method `isPending` that will check if the snapshot has one of the following values `Pending`, `InProgress`. 

I agree about the resolve reject. The other way would be to throw an error if the snapshot is still not idle

Reply:
> `isIdle` doesn't mean the snapshot is valid. It only means that it has settled from its last operation. We can also go the other way around and have a method `isPending` that will check if the snapshot is in one of the following states `PENDING`, `IN_PROGRESS`.
> 
> I agree about the resolve reject. The other way would be to throw an error if the snapshot is still not idle

Either way, I would name the method maybe `isComplete` or `isDone`
Idle I interpret this as "there's something that will go on, but not right now"

Reply:
`isDone` it is.